telcell/celldb/pgdatabase.py

```diff
@@ -151,21 +163,33 @@ def search(
                 qwhere.append(
                     f"NOT ST_DWithin(rd, 'SRID=4326;POINT({x} {y})', {distance_lower_limit_m})"
                 )
+
         if date is not None:
             qwhere.append("(date_start is NULL OR %s >= date_start)")
             qwhere.append("(date_end is NULL OR %s < date_end)")
             qargs.extend([date, date])
 
-        if radio is not None:
-            if isinstance(radio, str):
-                radio = [radio]
-            qwhere.append(f"({' OR '.join(['radio = %s'])})")
-            qargs.extend(radio)
+        if ci is not None:
+            add_qwhere, add_qargs = _build_cell_identity_query(ci)
+            qwhere.append(add_qwhere)
+            qargs.extend(add_qargs)
 
-        if mcc is not None:
-            qwhere.append(f"mcc = {mcc}")
-        if mnc is not None:
-            qwhere.append(f"mnc = {mnc}")
+            assert radio is None, "radio argument makes no sense in combination with ci"
+            assert mcc is None, "mcc argument makes no sense in combination with ci"
+            assert mnc is None, "mnc argument makes no sense in combination with ci"
+        else:
+            if radio is not None:
+                if isinstance(radio, str):
+                    radio = [radio]
+                qwhere.append(f"({' OR '.join(['radio = %s'])})")
+                qargs.extend(radio)
```

Comment:
These seem like sql clauses to be string-formatted into a query later; this makes me very anxious, is there really no other way to do this? :see_no_evil:

Reply:
Nope, these are sql (postgres) substitutions, not to be formatted in python